relay-quotas/src/is_rate_limited.lua

```diff
@@ -28,24 +29,39 @@
 -- The result is a Lua table/array (Redis multi bulk reply) that specifies
 -- whether or not the item was *rejected* based on the provided limit.
 assert(#KEYS % 2 == 0, "there must be 2 keys per quota")
-assert(#ARGV % 3 == 0, "there must be 3 args per quota")
-assert(#KEYS / 2 == #ARGV / 3, "incorrect number of keys and arguments provided")
+assert(#ARGV % 4 == 0, "there must be 4 args per quota")
+assert(#KEYS / 2 == #ARGV / 4, "incorrect number of keys and arguments provided")
+
+-- parse the incoming string into boolean, returns `nil` if could not parse which is also considered `false`
+local function parse_boolean(v)
```

Comment:
Can we agree to a protocol and not do these operations? They run quite often and it's expensive. For example, we could use `t` as truthy value and consider it falsy otherwise.

Reply:
Good idea, we could check for the string `"true"` instead (assuming that is what the rust redis library converts the argument to).

Reply:
I reduced this only to `"true"`  and `"false"` but it does not really matter, we still have to have comparison to the string to get the boolean from that, either we assign this to the variable or do it in the if, there still will be that additional operation.  IMHO  having function and variable is much cleaner.

Reply:
go lisp-style and use `t` for faster `strcmp` results :)

i'm not sure what the overheads are of a function in lua, but it is used only once to assign to a variable (which i agree is clearer to read) so could as well write the if statement right there.